### DISCUSSION TITLE
Is there a way to define conditional connections in the flow?

### DISCUSSION THREAD

**Discussion by amichel** [microsoft/promptflow#3669] [General]
Is there a way to define connection as the flow input?
I want to select a connection based on the input and then use it in the nodes, so if I need to change to a different connection, I will only do it once in the flow, not in every node
Currently, it only seems possible to use one of the existing connections as the node input in the connection element.

I understand that I can pass the connection name as a string and then use `client.connections.get` utility in every downstream tool, but it would be more intuitive to allow using strongly typed connections in tool parameters and define it in the beginning

**Comment by brynn-code** [comment]
How about define the two connections as inputs at the same time? And the tool logic choose to use the one required.

**Comment by amichel** [comment]
I see what you mean. It can work with static cases indeed, when all
connections are known at design time.

**Reply by brynn-code** [reply]
I see the point, so we don't know all the connections... I'm afraid we don't support such conditional parameters, sorry about that.
